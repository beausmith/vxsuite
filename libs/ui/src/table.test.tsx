import React from 'react';
import { render } from '@testing-library/react';

import { Table, TD } from './table';

describe('renders Table', () => {
  test('as table tag', () => {
    const { container } = render(<Table />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('can specify borderTop', () => {
    const { container } = render(
      <Table borderTop>
        <tbody>
          <tr>
            <td>Willow</td>
          </tr>
        </tbody>
      </Table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('can specify condensed', () => {
    const { container } = render(
      <Table condensed>
        <tbody>
          <tr>
            <td>Evermore</td>
          </tr>
        </tbody>
      </Table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('can render TD inside Table', () => {
    const { container } = render(
      <Table condensed>
        <tbody>
          <tr>
            <td>Willow</td>
            <TD>Daylight</TD>
            <TD narrow textAlign="center">
              Renegade
            </TD>
            <TD nowrap>Cardigan</TD>
            <TD narrow nowrap textAlign="right">
              Seven
            </TD>
          </tr>
        </tbody>
      </Table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
