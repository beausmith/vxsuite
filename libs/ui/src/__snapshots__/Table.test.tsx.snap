// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Table as table tag 1`] = `
<table
  class="sc-bdvvaa NPNLJ"
/>
`;

exports[`renders Table can render TD inside Table 1`] = `
<table
  class="sc-bdvvaa jWShPI"
>
  <tbody>
    <tr>
      <td>
        Willow
      </td>
      <td
        class="sc-gsDJrp fbuCsp"
      >
        Daylight
      </td>
      <td
        class="sc-gsDJrp chLbkq"
      >
        Renegade
      </td>
      <td
        class="sc-gsDJrp UTrqs"
      >
        Cardigan
      </td>
      <td
        class="sc-gsDJrp sHCCM"
      >
        Seven
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders Table can specify borderTop 1`] = `
<table
  class="sc-bdvvaa eYNfPV"
>
  <tbody>
    <tr>
      <td>
        Willow
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders Table can specify condensed 1`] = `
<table
  class="sc-bdvvaa jWShPI"
>
  <tbody>
    <tr>
      <td>
        Evermore
      </td>
    </tr>
  </tbody>
</table>
`;
